enhancements/etcd/storage-migration-for-etcd-encryption.md

```diff
@@ -0,0 +1,122 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Introduce the `openshift-storage-migration-operator` to install the `StorageVersionMigration` CRD and the corresponding  `kube-storage-version-migrator` controller.  `StorageVersionMigration` resources can then be created by components, (initially  `kube-apiserver` and `openshift-apiserver`) to kick off storage migrations for resources.
+
+The first known usage is for etcd encryption at rest: `secrets` (`kube-apiserver`) and `oauthaccesstokens` (`openshift-apiserver`).
+
+## Motivation
+
+The immediate need for storage migration is to support the enablement of etcd data storage encryption. Encrypting etcd data storage involves rotating encryption keys periodically, which requires that stored data be migrated after choosing a new write key and before removing an old read key so that api servers can discard old encryption keys periodically and not lose the ability to decode the pre-existing data.
+
+We will use the upstream `kube-storage-version-migrator` (the controller-based successor to `oc adm migrate`). While upstream usage of the `kube-storage-version-migrator` is focused on migrating to new storage versions, we should be able to explicitly trigger the `kube-storage-version-migrator` when we need to migrate the encryption key used for storage, even when the storage version itself has not changed.
+
+## Goals
+
+* Create a `openshift-storage-migration-operator`, managed by the CVO, to manage the resources needed to run the `kube-storage-version-migrator`.
+* Initiate a storage migration of resources, even if the storage version has not changed.
+* Monitor the progress of storage migration of resources.
+* Enable the kube-apiserver to utilize the `kube-storage-version-migrator` when managing etcd encryption keys.
+* Inform the upstream `kube-storage-version-migrator` of our usage and our success/failure to gain influence for future decisions and create tests to ensure our use-cases.
+
+## Non-Goals
+
+* Will not provide a higher level API to the `StorageVersionMigration` api controlled by the `kube-storage-version-migrator`.
+* Will not implement any general control loops to automatically detect and then initiate a storage migration when needed.
+
+## Proposal
+
+* Introduce `openshift/cluster-openshift-storage-migration-operator` repository.
+  * Not required for bootstrapping or re-bootstrapping.
+  * No need for static pods.
+  * Will run on cluster.
+* Implement `openshift-storage-migration-operator`. 
+* Build `kube-storage-version-migrator` image.
+  * Fork `kubernetes-sigs/kube-storage-version-migrator` to `openshift`
+  * Enable CI on `openshift/kube-storage-version-migrator` repo.
+* Utilize `StorageVersionMigration` api when etcd storage encryption keys are updated.
+
+## User Stories [optional]
+
```

Comment:
these you have in the Goals section already.

Reply:
Updated